consensus/phases/thirdphase.go

```diff
@@ -129,6 +138,31 @@ func (tp *ThirdPhaseImpl) Execute(ctx context.Context, pulse *core.Pulse, state
 	}, nil
 }
 
+func (tp *ThirdPhaseImpl) removeExcessJoinClaims(joinClaims []*packets.NodeJoinClaim, ref core.RecordRef, state *SecondPhaseState) {
+	claims := state.UnsyncList.GetClaims(ref)
+	originLen := len(claims)
+	if originLen == 0 || len(joinClaims) == 0 {
+		return
+	}
+	updatedClaims := make([]packets.ReferendumClaim, 0)
+	for _, join := range joinClaims {
```

Comment:
Filter `joinClaims` before this loop to have only claims from node `ref`. If the filtered list of join claims is empty, you can early return